app/Filament/Resources/GameResource/RelationManagers/AchievementsRelationManager.php

```diff
@@ -0,0 +1,238 @@
+                                }
+
+                                $record->type = AchievementType::Progression;
+                                $record->save();
+                            });
+                        }),
+
+                    Tables\Actions\BulkAction::make('type-win-condition')
+                        ->label('Set selected to Win Condition')
+                        ->color('success')
+                        ->requiresConfirmation()
+                        ->action(function (Collection $records) use ($user) {
+                            $records->each(function (Achievement $record) use ($user) {
+                                if (!$user->can('updateField', [$record, 'type'])) {
+                                    return;
+                                }
+
+                                $record->type = AchievementType::WinCondition;
+                                $record->save();
+                            });
+                        }),
+
+                    Tables\Actions\BulkAction::make('type-missable')
+                        ->label('Set selected to Missable')
+                        ->color('warning')
+                        ->requiresConfirmation()
+                        ->action(function (Collection $records) use ($user) {
+                            $records->each(function (Achievement $record) use ($user) {
+                                if (!$user->can('updateField', [$record, 'type'])) {
+                                    return;
+                                }
+
+                                $record->type = AchievementType::Missable;
+                                $record->save();
+                            });
+                        }),
+
+                    Tables\Actions\BulkAction::make('type-null')
+                        ->label('Remove type from selected')
+                        ->requiresConfirmation()
+                        ->action(function (Collection $records) use ($user) {
+                            $records->each(function (Achievement $record) use ($user) {
+                                if (!$user->can('updateField', [$record, 'type'])) {
+                                    return;
+                                }
+
+                                $record->type = null;
+                                $record->save();
+                            });
+                        }),
+                ])
+                    ->label('Bulk set type')
+                    ->visible(fn (): bool => $user->can('updateField', [Achievement::class, null, 'type'])),
+            ])
+            ->recordUrl(
+                fn (Achievement $record): string => route('filament.admin.resources.achievements.view', ['record' => $record])
+            )
+            ->paginated([50, 100, 'all'])
+            ->defaultPaginationPageOption(50)
+            ->defaultSort('DisplayOrder')
+            ->reorderable('DisplayOrder', $this->canReorderAchievements())
```

Comment:
DisplayOrder changes don't appear in the audit log.

Reply:
Locally, I made a change to include these diffs in the log. However, it seems when the value is changed via Filament's `->reorderable()` (drag and drop), it isn't recorded at all.

Filament must be bypassing the conventional Eloquent ORM methods to update these values. Even attempting to update `Achievement::boot()`'s `static::updated()` function to log these changes manually was not successful.

Next, I investigated if Filament supports some sort of callback function for when a reorderable save happens, hoping to perhaps manually create a log record there. I did find (and used) this functionality, but its diffing capabilities are extremely limited.

My perception is the intent of logging changes here is to be able to track down a rogue dev who might be altering these display order values on sets. I think the solution I've landed on should cover that use case.

Where I've ultimately landed is:
* When a user reorders achievements, a basic event is recorded to the activity log.
* If they continue reordering achievements within a 10-minute timespan, events will not be logged. There is a basic throttle in place to mitigate log noise.

These logs are bare minimum and look like this:
![Screenshot 2024-07-17 at 8 07 08 PM](https://github.com/user-attachments/assets/9fdc2577-2b16-49cd-844e-59e6473f2283)

Reply:
I don't seem to have an activity log.
> Base table or view not found: 1146 Table 'retroachievements-web.activity_log' doesn't exist

Reply:
These records should be written to the `audit_log` table:

<img width="1454" alt="Screenshot 2024-07-17 at 11 20 35 PM" src="https://github.com/user-attachments/assets/d069e63a-ea72-4497-9d52-58bd1b0a428b">


In _config/activitylog.php_, I have:
```php
    'table_name' => 'audit_log',
```

Does the error you're encountering occur when you open the activitylog page, or does it happen when you try to reorder an achievement?

Reply:
The error occurs after the drop of the drag-and-drop while reordering.

I also have "table_name=>audit_log" in the config/activitylog.php file.

The code is explicitly looking for activity_log:
```
        // We don't want to flood the logs with reordering activity.
        // We'll throttle these events by 10 minutes.
        $recentReorderingActivity = DB::table('activity_log')
            ->where('causer_id', $user->id)
```

Reply:
Ah - I didn't notice I was on the wrong branch and missed my typo. Resolved in latest.

Reply:
> If they continue reordering achievements within a 10-minute timespan, events will not be logged. There is a basic throttle in place to mitigate log noise.

I don't seem to be limited to 10 minutes per log:
![image](https://github.com/user-attachments/assets/bffaa96f-990e-447b-96e6-07975a0e13ac)

Reply:
Fixed in latest - `Game::class` was working fine in my REPL tool, but not in Filament itself.